src/CsvHelper/Expressions/RecordWriter.cs

```diff
@@ -86,13 +86,44 @@ protected Action<T> GetWriteDelegate<T>(T record)
 			return (Action<T>)action;
 		}
 
+		/// <summary>
+		/// Gets the delegate to write the given record. 
+		/// If the delegate doesn't exist, one will be created and cached.
+		/// </summary>
+		/// <typeparam name="T">The record type.</typeparam>
+		/// <param name="typeInfo">The type for the record.</param>
+		protected internal Action<T> GetWriteDelegate<T>(RecordTypeInfo typeInfo)
+		{
+			var typeKey = typeInfo.RecordType.GetHashCode();
+
+			if (typeInfo.IsItemType)
+				typeKey = HashCode.Combine(17180427, typeKey);

```

Comment:
What is the purpose of this? In the case of `object`, `typeInfo.RecordType` was already retrieved via `GetType()`.